### ISSUE TITLE
Product-to-test doc citations are unverifiable prose, and have been wrong in three consecutive reviews

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#3641]
## Problem

Product doc comments cite the test that gates a claimed safety property, per the AGENTS.md
rule that "a doc comment or README that asserts a safety, soundness, or faithfulness
property must name the gate that enforces it."

A `<see cref="..."/>` is compiler-checked. But a product assembly cannot reference its test
assembly, so every product-to-test citation degrades to plain `<c>TestName</c>` — a string
no tool checks. Nothing verifies that:

1. the named test exists,
2. the named test is the one that actually gates the property.

## Evidence this is not hypothetical

During review of #3580 the same citation was wrong in three consecutive rounds:

- **Round 4** — flagged, and dismissed by me: I verified the *behavior* was correct and
  never checked the *citation*.
- **Round 6** — flagged again, and I corrected the one occurrence that was reported.
- **Round 7** — flagged a third time. The claim appears in three places; only one had been
  fixed, so the product file kept carrying it.

The claim was that `ForgedKeySegmentation_DoesNotFoldAcrossDeclaringTypes` gates the key
flattening's injectivity. Measured by setting `KeySeparator` to a spellable `'@'`:

```text
Total: 196, Errors: 0, Failed: 1
    ILInspector.Instructions.Tests.CompilerGeneratedOrdinalTests.KeySeparatorCannotBeSpelledByAMetadataName [FAIL]
```

`ForgedKeySegmentation_...` does not fail. It is written against the historical
`.`/`+`/`::` joining, so it exhibits the hazard but does not gate the general claim.

Both the cited test and the correct test exist, so a mere existence check would have caught
neither. Both citations were in `<c>...</c>` form, so the compiler saw nothing.

## Why it matters

An incorrect gate citation is worse than no citation. It reads exactly like a verified
property and it tells the next reader — and the next reviewer — that tampering with the
mechanism is already covered. The only thing that reliably distinguishes a real gate from a
plausible one is tampering with the mechanism and reading which test goes red, which is
manual and does not survive refactoring.

## Options

1. **Existence gate (cheap, partial).** A test that scans the product source for
   `<c>Identifier_WithUnderscores</c>` citations and asserts each names a `[Fact]`/`[Theory]`
   in the test assembly. Catches renamed and deleted tests; catches neither defect above.
   Note this reintroduces source-text scanning, which #3580 deliberately replaced with a
   runtime check in another context — the purposes differ, but the precedent is worth
   weighing.

2. **Invert the direction (better).** Keep the *claim* and its citation in the test file,
   where `<see cref="..."/>` is compiler-checked, and have the product comment point at the
   test class as a whole. Costs some locality in the product file.

3. **Make the gate executable (best, most expensive).** A mutation-style harness that, for a
   registered `(mechanism, gating test)` pair, applies a declared mutation and asserts that
   exactly that test fails. This is what I have been doing by hand every round; the pairs are
   already written down in prose.

Option 2 is probably the right first step, and it composes with 3.

## Provenance

Found while reviewing #3580. Not blocking it — that PR's citations are now correct and
verified by tamper. Filing so the systematic cause is not lost with the individual fixes.
